A proposal for tool panel restructuring and tool review (#3806)
galaxyproject/tools-iuc: docs/tool_and_toolpane_guidelines.md

### Proposed guidelines

- Tool names should be as short and as expressive as possible 
- The bold portion of the tool name (defined by `name` attribute within `<tool>` tag) should be short 
- No parenthesis are allowed in tool names
- Only the first word is capitalized

### Examples

| Current name | Suggested edit |
|--------------|---------------|
|**Select first** lines from a dataset | **Select first** lines |
|**Sort** data in ascending and descending order | **Sort** data|
|**Samtools fastx** extract FASTA or FASTQ from alignment files | SAMTools should have a dedicated category within the toolpane (see **Tool categories** below). The suggested name then would be: "**fastx**: extract FASTA or FASTQ"|

## Help section style

Help section of the tools vary widely in content. However, just like we cannot accept tools without tests we should not be accepting tools with rudimentary tool sections. 

### Proposed guidelines

These guidelines partially inspired by `man` pages structure:

> - Each tool should include `Synopsis` section with a very brief description.

Reviewer: ```suggestion
- Each tool should include `Synopsis` section with a brief, one line phrase ending with a period. It prescribes what the tool does and is not a tool description.
```

- `Synopsis` should be followed by `Description` section
- If this is a homegrown tool, the `Description` section can be free form
- If this is an external package (e.g., SAMTools), the `Description` section should be a copy of help section for this tool adopted for Galaxy. This means that it should only describe CLI flags that are present within `<inputs>` section.
- When appropriate help section should include images preferably in `svg` format.
- For complex tools `<help>` section should include examples.

### Examples

Before:

![](https://i.imgur.com/oV4uY5m.png)

After:

![](https://i.imgur.com/G6rxKq3.png)

## Tool Categories

Tool section needs to be completely reworked. The existing section dates back to 2012/2013 and no longer reflects the breadth of Galaxy's toolset. 

### Proposed toolpane structure

Here top level sections (e.g., `Get Data`) are labels such as, for example, **GENERAL TEXT TOOLS** on usegalaxy.org. The next level are expandable tool categories containing actual tools. One open question is how to handle tool from different disciplines (e.g., genomics vs. climate science).

- Get data
    - NCBI SRA
    - NCBI Datasets
    - UCSC Table Browsers

>     - EBI

Reviewer: Each section should be alphabetically sorted.

> Here top level sections (e.g., `Get Data`) are labels such as, for example, **GENERAL TEXT TOOLS** on usegalaxy.org. The next level are expandable tool categories containing actual tools. One open question is how to handle tool from different disciplines (e.g., genomics vs. climate science). Within each section categories and tools must be alphabetically listed.

Reviewer 2: This whole section is misplaced in the IUC or IUC standards, we can't enforce this, and this won't be available to tool installers or Galaxy instances, and this isn't part of the tool config file. EDAM terms are part of tool config files, so that seems like the logical choice to me. Are you opposed to mapping EDAM ontologies to sections in a config file in Galaxy ?

This can be shared across usegalaxy.* instances in a fairly automatic way.
It also provides a means for subcategories to appear multiple times in the tool panel (see the points raised by @bgruening), so we can have for instance the statistics tools appear in a statistics section as well as in more specialized sections like ML or Differential expression.

Author: I'm not opposed but EDAM categories can be equally confusing. Should IUC require EDAM categories as merge criterium?

Reviewer 3: We can not make it a requirement as EDAM is not complete and the Galaxy tools are more diverse than EDAM is today. However we can make this a strong recommendation.

Reviewer 4: We could have a supplementary IUC ontology for the gaps? And require it's in one of the two with the IUC one being more of a "testbed" ontology and having the EDAM folks take inspiration for new categories from us?

Like @mvdbeek I'm a huge fan of the way EDAM categories are automatic and would save us admins a large amount of time, and improve discoverability by the multiple listings.

(But I get the point of @nekrut, he wants a samtools category because his users expect that name, something like that we wouldn't necessarily get with edam)

Reviewer 5: > We could have a supplementary IUC ontology for the gaps? And require it's in one of the two with the IUC one being more of a "testbed" ontology and having the EDAM folks take inspiration for new categories from us?

I think EDAM would welcome us contributing missing categories.

- Interactive tools
    - Notebook environments
    - Other IEs
- Dataset collections
    - Collection operations
- Text and dataframes
    - Filtering [filters of all sorts]
    - Find/Replace [sed ...]
    - Column manipulation [cut1, cut2, paste, merge...]
    - Sorting [sort]
    - Join, Subtract, Group
    - Misc text tools

>     - Statistical operations

Reviewer 3: should statistics really be here or a separate section? Esp. in comparison to the ML extra section.

Author: indeed. new section with ML

    - Datamash
    - IWTomics
- Format conversion
    - FASTA/FASTQ
    - BED, GFF

> - FASTA/FASTQ

Reviewer 3: FASTA/FASTQ and Genomic Interval does not feel like the rest.

What about something like "Sequence manipulation and QC"?

Should this section also contain FAST5 tools?

Author: Yes, I think Sequence manipulation and QC would be better. And yes, FAST5 should be there as well in a separate category

    - FASTA manipulation
    - SeqTk
    - FASTQ manipulation
    - QC
    - EMBOSS
- Genomic intervals
    - BEDTools
- Read mapping
    - Short read data
    - Long read data
- Mapped data
    - SAMTools
    - Picard

>     - BEDTools [can we list same category twice?]

Reviewer: I'd suggest singleton category names.

    - DeepTools
    - QualiMap
    - UMI-tools
- Assembly
    - Viral and bacterial
    - Animals and plants
    - Annotation tools
    - VGP tools
- Metagenomics
    - Kraken
    - dada2
    - Mothur
    - VSearch
    - Diamond
    - Vegan
    - Diamond
    - CAT
    - Misc 
- Variation data
    - VCFTools
    - BCFTools
    - SNPEff
    - Diploid callers
    - Mixed/Haploid callers
    - MiModD
    - DuNovo

> - RNAseq

Reviewer: Consistency with ChIP-Seq: `RNA-seq`

> - Variation data

Reviewer 3: data seems to not fit in respect to the other names

Author: these are I think minor details. I hope we will not have an endless discussion about these ;)

    - Mapping
    - Transcript assembly
    - Quantification
    - SingleCell
    - RSeQC
-  ChIP-seq
    -  Peak calling
    -  Motif analysis
- Immunology
    - Presto
- Protein analysis
    - MS data analysis
    - Docking modeling
- Machine learning
    - [A ML expert will need to set categories here]

> - Computational Chemistry

Reviewer: Consistent capitalization with the rest of the panel and tools:
```suggestion
- Computational chemistry
```

> -  ChIP-seq

Reviewer 3: ```suggestion
- ChIP-seq
```

>     - Chemical Tool Box

Reviewer: ```suggestion
    - Chemical tool box
```

- Evolution

>     - Phylogenetic Tree Reconstruction

Reviewer: ```suggestion
    - Phylogenetic tree reconstruction
```

>     - Docking modeling

Reviewer 3: Docking is probably something that should go to Cheminformatics ... ping @simonbray

> - Protein analysis
>     - MS data analysis
>     - Docking modeling

Reviewer 6: ```suggestion
```

Reviewer 6: I don't think `Protein analysis` is very meaningful.

Reviewer 3: Would Proteomics be better?

Author: Proteomics it is!

>     - Chemical tool box

Reviewer 6: ```suggestion
    - Cheminformatics
    - Protein-ligand docking
    - Molecular dynamics
```
